libvast/src/store.cpp

```diff
@@ -222,8 +222,8 @@ base_store::count(expression expr, ids selection) const {
 detail::generator<table_slice>
 base_store::extract(expression expr, ids selection) const {
   for (const auto& slice : slices()) {
-    if (auto filtered_slice = filter(slice, expr, selection))
-      co_yield std::move(*filtered_slice);
+    for (auto selected : select(slice, expr, selection))
+      co_yield std::move(selected);
```

Comment:
This is the one change I would like to get @tobim's input on. Everything else is mostly trivial, shuffling code around, or already reviewed by @mavam.

I'd also be happy to move this into a follow-up PR as this likely has performance implications.

Reply:
I benchmarked this on my laptop and discovered that `filter` performs a lot better for queries that hit lots of small runs in the data. `select` is actually up to 25x times slower. Let's revert this change, the rest is good to go!

Reply:
I rebased onto master to resolve the merge conflict and dropped the one commit that made this change (it changed nothing else). Thanks for testing this extensively!